pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1774,6 +1777,41 @@ func (c *Cloud) DiskIsAttached(diskName string, nodeName types.NodeName) (bool,
 	return false, nil
 }
 
+func (c *Cloud) DisksAreAttached(diskNames []string, nodeName types.NodeName) (map[string]bool, error) {
+	attached := make(map[string]bool)
+	for _, diskName := range diskNames {
+		attached[diskName] = false
+	}
+	awsInstance, err := c.getAwsInstance(nodeName)
+	if err != nil {
+		if err == cloudprovider.InstanceNotFound {
+			// If instance no longer exists, safe to assume volume is not attached.
+			glog.Warningf(
+				"Node %q does not exist. DisksAreAttached will assume disks %v are not attached to it.",
+				nodeName,
+				diskNames)
+			return attached, nil
+		}
+
+		return attached, err
+	}
+	info, err := awsInstance.describeInstance()
+	if err != nil {
+		return attached, err
+	}
+	for _, blockDevice := range info.BlockDeviceMappings {
+		volumeId := aws.StringValue(blockDevice.Ebs.VolumeId)
+		for _, diskName := range diskNames {
+			if volumeId == diskName {
```

Comment:
If you want, we can merge this as is, and then I might do the same thing I did with `types.NodeName`, with `aws.VolumeNameSpec` or something, so I can catch any more of these.

I don't think we gate on AWS e2e failures?